src/classes/UTIL_Describe.cls

```diff
@@ -112,6 +113,36 @@ public class UTIL_Describe {
             }
         }
     }
+	
+    /*******************************************
+    * Gets a list of picklist entries for a new object/field/list<PicklistEntry> triple
+    ********************************************/
+    private static void fillPicklistEntriesForObject(string objectName, string fieldName) {
+		
+		// make sure we have this field's schema mapped
+        if (!fieldDescribes.containsKey(objectName) || !fieldDescribes.get(objectName).containsKey(fieldName))
+            fillFieldMapsForObject(objectName, fieldName);
+				
+		// Check for picklist.
+		if(getFieldType(objectName,fieldName) != 'PICKLIST') {
+			throw new SchemaDescribeException('Invalid Selection. Field name \'' + fieldName + '\' for object name \'' + objectName + '\' is not a picklist.');
+		}
+
+		// fill entries for the first time.
```

Comment:
Is this comment needed, it just seems to be floating there?

Reply:
I pulled the comment out. The other comments cover the exact behavior now.